### ISSUE TITLE
Large directories sizes are not correct (>10 GB)

### ISSUE THREAD

**Issue by joel-mag38** [linuxmint/nemo#3495]
### Distribution

Mint 22

### Package version

6-2-8

### Frequency

Always

### Bug description

I copied a large directory (200 GB) named A from external HDD (WD Elements 2TB) to home  named B. Then I copied B to a directory named C on an other external drive. Then I copied directly A to the other external drive directory named D. 
The sizes of A, B,C et D are all different.
I checked the sizes with Windows. The sizes of A, C, D are all the same, with a value different from the size of B

### Steps to reproduce

each time I read the size of a directory, I get the same value.

### Expected behavior

The copy is good, the sizes should be identical.

### Additional information

In fact, sizes are not correct for directories larger than 1GB (not the same value than Windows)
My directory contains 47000 elements.

**Comment by claudiux** [comment]
Are you sure using the same unit in both systems? 1GB = 1 024 or 1 000 MB?

**Comment by joel-mag38** [comment]
Sizes are in bytes. For smaller  sizes, there is no problem

**Comment by joel-mag38** [comment]
On the 2 external drives, NTFS. On linux, I used classic installation, I
choosed to encrypt the home directory.  The home directory was on a special
drive.
For the sizes, I clicked right, and choosed properties .

**Comment by joel-mag38** [comment]
Sizes
A 222726304291
B 222724370979
C 222725968419
D 222726128163

Size ACD windows 222715617827

**Comment by joel-mag38** [comment]
I have no hidden files, only photos.
Directory Size is only the sum of file sizes, not the space used.
For smaller directories, less than 1 GB, I have no problem.

**Comment by Jeremy7701** [comment]
Since you copied A to D directly (both NTFS) it seems strange that they are not quite the same size.

**Comment by joel-mag38** [comment]
The copy is made with linux.
I only used windows to check the sizes, because I was surprised.

**Comment by Jeremy7701** [comment]
Linux can handle NTFS. I suspect the reason is down to filesystem structures such as (for example) the MFT.

**Comment by joel-mag38** [comment]
I am surprised because I used to check good copies in checking the sizes.
I am not sure Linux sizes are good.  I will go through the problem later .
Thank you

**Comment by joel-mag38** [comment]
Hi,
I have looked a little at directory sizes.
I think directories sizes should be equal at the sum of file sizes  + the
sum of subdirectory sizes.
For a partition linux, ext4, the subdirectory size is equal to 4096 bytes,
maybe a block size...
For a partition NTFS, it is equal to zero.
a file size on ext4 is equal to a file size on NTFS.
All this works well for directories with small numbers of elements.
ext4 directories sizes are equal to NTFS directories sizes if there are no
subdirectories.

However, for directories with large numbers of elements, there are problems.

ext4 directories sizes are larger than the sum of directories sizes + the
sum of files sizes, this difference is a multiple of a block size.

I have copied an ext4 directory to the same partition, and the sizes of
directories are different of 2 block sizes.
57 200 703 583
57 200 695 391

for NTFS directories sizes, it is the same,
NTFS directories sizes are larger than the sum of file sizes, and this
difference is a multiple of a block size.

I hope this will help, maybe it is due to my hardware, I don' t know.

Thank you for the work you are doing in developing Linux.

Joël Magnin
